### ISSUE TITLE
Final Cluster != N_components in the estimated GMM density ???

### ISSUE THREAD

**Issue by dominikusbrian** [anniewesterlund/Tutorial_FE_clustering#2]
Dear Dr. Westerlund,

I recently discover this InfleCS technique and tested it for finding cluster centres for my system. 
I found this is a great contribution to the community indeed. 
However, I am rather confused why that the model that performs clustering with GMM density n_components of e.g. 10 or 16, produce cluster less than that amount, in my case, it is 5, and 13, respectively. 

Looking forward to your insight on this issue, 
Best regard,
Domi.

**Comment by anniewesterlund** [comment]
Hi Domi,

I am happy that you found InfleCS, and hopefully the jupiter notebook tutorial in delemottelab/InfleCS-free-energy-clustering-tutorial, useful!

The reason why the clustering results in fewer clusters than components is because the GMM tries to fit a density as accurately as possible, while the clustering extracts peaks from the density. So if you have more n_components than clusters, it means that more-than-one component was used to fit one density peak, e.g. if you have a non-Gaussian peak this would be the case. Then, the clustering essentially merges components since we are interested in the number of density peaks in the end. Hope this makes things clearer!

**Comment by dominikusbrian** [comment]
Hi Annie,

Thank you for your explanation. Yes, the Jupyter tutorial is really useful, especially that it is written in a clear and organized fashion. I see now why there's less cluster than components. That makes things even more interesting because in [my previous work ](https://pubs.acs.org/doi/10.1021/acs.jctc.0c01250) we did found out that for our system there is some sort of non-gaussian nature in the response function obtained from the conformation ensemble of our molecule. This is also in agreement with the not-so Gaussian distribution of the descriptor I currently use for obtaining the free energy landscape. 

Following up on this question, I am wondering if there is some way to know which particular components were merges with which when the InfleCS algorithm use them to fit for the final number of density peaks? or is there a way to restrict this behaviour? So that one could compare the prediction of core state if all n_components were considered explicitly vs when multiple components were used to fit one density peak.

**Comment by anniewesterlund** [comment]
Hi! Interesting. 

1. One thing that you could do to see which components were merged is to check which clusters the component means belong to. The GMM density is stored in 
`fec.density_est_`
So the means are found with `fec.density_est_.means_`
You can then use the 
`labels = fec.evaluate_clustering(fec.density_est_.means_, assign_transition_points=False)`
to obtain the cluster belongings.
2. If you want to compare to the ordinary GMM clustering, you could obtain that with the `fec.density_est_.predict(X)` function (see https://scikit-learn.org/stable/modules/generated/sklearn.mixture.GaussianMixture.html). If you only want to consider the core state points, you could just evaluate them in the `fec.density_est_.predict(X)` and keep the rest as transition points.

**Comment by dominikusbrian** [comment]
Fantastic! 

Okay, everything is good. I can now proceed by comparing the InfleCS with the other result I have from other clusterings approach. Yes, I do have result from the ordinary GMM clustering, but the core state and transition points separated data are even better. 

Thanks, Annie for the insights on details about your InfleCS method. 
Nice Work~
